### PR TITLE
Remove capella types

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Removing capella types as part of clean up after the Deneb hard fork. Note that due to an SSZ encoding bug in the attestant library for the deneb builder bid, it has been updated to the latest version.

## ⛱ Motivation and Context

This allows the mev-boost codebase to be more manageable instead of accumulating more code from previous hard forks.

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`

### CODE REVIEW

File: server/mock/mock_types_test.go [hunk 1]
```diff
@@ -205,9 +205,10 @@ func TestHexToSignature(t *testing.T) {
 
 	publicKey := hexutil.Encode(bls.PublicKeyToBytes(blsPublicKey))
 
-	message := &builderApiCapella.BuilderBid{
-		Header: &capella.ExecutionPayloadHeader{
-			BlockHash: HexToHash("0xe28385e7bd68df656cd0042b74b69c3104b5356ed1f20eb69f1f925df47a3ab7"),
+	message := &builderApiDeneb.BuilderBid{
+		Header: &deneb.ExecutionPayloadHeader{
+			BlockHash:     HexToHash("0xe28385e7bd68df656cd0042b74b69c3104b5356ed1f20eb69f1f925df47a3ab7"),
+			BaseFeePerGas: uint256.NewInt(0),
```
Comment (MariusVanDerWijden) [inline server/mock/mock_types_test.go:211]: Why did you add BaseFeePerGas here?

**Review by MariusVanDerWijden — APPROVED** [review]
Sounds good, we should get that in soonish so I can rebase my PRs on top

File: server/mock/mock_types_test.go (hunk 1 above)
Comment (avalonche) [reply server/mock/mock_types_test.go:211]: `BaseFeePerGas` no longer can be nil in the new `deneb.ExecutionPayloadHeader` type

**Review by jtraglia — APPROVED** [review]
